pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperKitCreationService.java

```diff
@@ -0,0 +1,276 @@
+import org.broadinstitute.dsm.model.KitType;
+import org.broadinstitute.dsm.util.DDPKitRequest;
+import org.broadinstitute.dsm.util.EasyPostUtil;
+import org.broadinstitute.lddp.db.SimpleResult;
+import org.broadinstitute.lddp.util.DeliveryAddress;
+
+@Slf4j
+public class NonPepperKitCreationService {
+    public static final String JUNIPER = "JUNIPER";
+    public static final String JUNIPER_UNDERSCORE = "JUNIPER_";
+
+    //These are the Error Strings that are expected by Juniper
+    public static final String ADDRESS_VALIDATION_ERROR = "UNABLE_TO_VERIFY_ADDRESS";
+    public static final String UNKNOWN_KIT_TYPE = "UNKNOWN_KIT_TYPE";
+    public static final String UNKNOWN_STUDY = "UNKNOWN_STUDY";
+    public static final String MISSING_JUNIPER_KIT_ID = "MISSING_JUNIPER_KIT_ID";
+    public static final String MISSING_JUNIPER_PARTICIPANT_ID = "MISSING_JUNIPER_PARTICIPANT_ID";
+    public static final String DSM_ERROR = "DSM_ERROR_SOMETHING_WENT_WRONG";
+
+    public KitResponse createNonPepperKit(JuniperKitRequest juniperKitRequest, String studyGuid, String kitTypeName) {
+        if (StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID())) {
+            return new KitResponse(MISSING_JUNIPER_PARTICIPANT_ID, juniperKitRequest.getJuniperKitId(),
+                    juniperKitRequest.getJuniperParticipantID());
+        }
+        if (StringUtils.isBlank(juniperKitRequest.getJuniperKitId())) {
+            return new KitResponse(MISSING_JUNIPER_KIT_ID, null, juniperKitRequest.getJuniperKitId());
+        }
+        //getting the instance with isHasRole being set to true if the instance has role juniper_study
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceWithRoleByStudyGuid(studyGuid, "juniper_study");
+        if (ddpInstance == null) {
+            log.error(studyGuid + " is not a study!");
+            return new KitResponse(UNKNOWN_STUDY, juniperKitRequest.getJuniperKitId(), studyGuid);
+        }
+        if (!ddpInstance.isHasRole()) {
+            log.error(studyGuid + " is not a Juniper study!");
+            return new KitResponse(UNKNOWN_STUDY, juniperKitRequest.getJuniperKitId(), studyGuid);
+        }
+        HashMap<String, KitType> kitTypes = KitType.getKitLookup();
+        String key = KitType.createKitTypeKey(kitTypeName, ddpInstance.getDdpInstanceId());
+        KitType kitType = kitTypes.get(key);
+        if (kitType == null) {
+            return new KitResponse(UNKNOWN_KIT_TYPE, juniperKitRequest.getJuniperKitId(), kitTypeName);
+        }
+
+        Map<Integer, KitRequestSettings> kitRequestSettingsMap =
+                KitRequestSettings.getKitRequestSettings(String.valueOf(ddpInstance.getDdpInstanceId()));
+        KitRequestSettings kitRequestSettings = kitRequestSettingsMap.get(kitType.getKitTypeId());
+
+        // if the kit type has sub kits > like for testBoston
+        //        boolean kitHasSubKits = kitRequestSettings.getHasSubKits() != 0;
```

Comment:
Is this code still relevant?

Reply:
I want to keep the subkit code in there so in any case if anyone wants to add them in neaqr future there is a clue of where to begin